nim-compile.el

```diff
@@ -126,28 +105,26 @@ The config file would one of those: config.nims, PROJECT.nim.cfg, or nim.cfg."
    (delq nil `(,nim-compile-command ,@args ,@nim-compile-user-args ,file))
    " "))
 
-(define-derived-mode nim-compile-mode compilation-mode "nim-compile"
-  "major-mode for nim compilation buffer.")
-
 ;;;###autoload
 (defun nim-compile ()
   "Compile and execute the current buffer as a nim file.  All output is writton into the *compilation* buffer."
   (interactive)
   (when (derived-mode-p 'nim-mode)
-    (add-hook 'compilation-filter-hook 'nim--colorize-compilation-buffer)
-    (add-hook 'compilation-finish-functions 'nim--remove-colorize-hook)
-    (nim-compile--set-compile-command)
-    (funcall 'compile compile-command 'nim-compile-mode)))
+    ;; (add-hook 'compilation-filter-hook 'nim--colorize-compilation-buffer)
+    ;; (add-hook 'compilation-finish-functions 'nim--remove-colorize-hook)
+    ;; (nim-compile--set-compile-command)
+    ;; (funcall 'compile compile-command 'nim-compile-mode)))
+    (call-interactively 'compile)))
 
 (require 'ansi-color)
 (defun nim--colorize-compilation-buffer ()
-  "Colorize compilation buffer."
+  "Colorize compilation buffer.  Add to `compilation-filter-hook', when you want to interpret terminal colors in output buffer."
   (let ((inhibit-read-only t))
     (ansi-color-apply-on-region compilation-filter-start (point-max))))
 
 (defun nim--remove-colorize-hook (_buf _process-state)
-  "Remove ‘nim--colorize-compilation-buffer’."
-  (when (get-buffer "*nim-compile*")
+  "Remove ‘nim--colorize-compilation-buffer’.  Currently unused."
+  (when (get-buffer "*compilation*")
```

Comment:
Wait sharing one emacs client with multiple users is a thing? Does that mean that multiple users edit the same buffer at the same time, but are forced to use the same configuration file?

Well the reason why I want to use the "abstract ``*compilation*`` buffer" is for me really much more like, "Why again do we need yet another derived buffer with yet another buffer name?". I don't think there is ever the reason to have more than one ``*compilation*`` buffer active. So whenever I am compiling, I would like to compile to that buffer. I like to keep things as simple as possible, and adding more buffers feels to me to add another layer, or burden or just things I don't need or want. I can be totally wrong here, and there can a hidden reason why you want to have yet so many different buffers. But I don't see the necessity to do anything except setting a default compilation command for the major mode.